app/controllers/health_profiles_controller.rb

```diff
@@ -1,48 +1,58 @@
 class HealthProfilesController < ApplicationController
 	before_action :require_login
+ health_profile
+
 	
+ master
 
 	def index
 	end
 
 	def new
+ 
+		@patient = Patient.find(params[:patient_id])
 		# @health_profile = current_user.patients.find(params[:id])
 		@health_profile = HealthProfile.new
 	end
 
 	def create
-		# @health_profile = current_user.health_profiles.new(health_profile_params)
-		@health_profile = HealthProfile.new
+health_profile
+		@patient = Patient.find(params[:patient_id])
+		@health_profile = HealthProfile.new(health_profile_params)
+		@health_profile.patient_id = current_user.id
```

Comment:
This doesn't affect the flow of the code and it's my fault because I kept referencing different ways of getting the patient's id, but maybe we can go through and standardize how we are fetching the user's ID. Right now we are creating `@patient = Patient.find(params[:id])`, but then on line 22 we are setting the `@health_profile.patient_id = current_user.id`. If somebody else were to review the code, it might be easier to follow if we use `@health_profile.patient_id = @patient.id` just so there aren't any question's raised about if there's supposed to be a difference between `current_user` and `@patient`